src/avram/model.cr

```diff
@@ -223,6 +224,14 @@ abstract class Avram::Model
     })
   end
 
+  macro setup_association_queries(associations, *args, **named_args)
+    {% for assoc in associations %}
+      def {{ assoc[:assoc_name] }}_query
+        {{ assoc[:type] }}::BaseQuery.new
+      end
+    {% end %}
+  end
```

Comment:
This is necessary because we don't have access to the types of the intermediary associations in a has many through